### PR TITLE
Fixed issue with variance aggregator calculating incorrect results in some circumstances.

### PR DESCRIPTION
**Pull request by fernst**
We've updated the formula to calculate variance.

Note that we keep a running mean and squared mean. To ensure overflow and/or catastrophic cancellation is not a problem, we divide by the count of aggregated records on each step.

We've validated that the outputs for this aggregator match calculations performed by the BigQuery engine.

### CODE REVIEW

**Review by sumengwang — APPROVED** [review]
LGTM

File: core-plugins/src/main/java/io/cdap/plugin/batch/aggregator/function/Variance.java [hunk 1]
```diff
@@ -63,20 +60,26 @@ public void mergeValue(StructuredRecord record) {
...
     mean += delta / count;
-    double delta2 = value.doubleValue() - mean;
-    squareMean += delta * delta2;
+
+    // Calculate delta of the squared value vs the squared mean and adjust the squared mean
+    double deltaSquared = valueSquared - squaredMean;
+    squaredMean += deltaSquared / count;
+
     // if we divide by count, the result will be population variance,
     // count - 1, we will get sample variance
```
Comment (sumengwang) [inline core-plugins/src/main/java/io/cdap/plugin/batch/aggregator/function/Variance.java]: this comment seems to be unnecessary now

**Comment by sanjanasandeep** [comment]
do we need unit tests for this?

**Review by sanjanasandeep — APPROVED** [review]
lgtm

**Review by sechegaray — APPROVED** [review]

File: core-plugins/src/main/java/io/cdap/plugin/batch/aggregator/function/Variance.java (hunk 1 above)
Comment (fernst) [reply core-plugins/src/main/java/io/cdap/plugin/batch/aggregator/function/Variance.java]: You're right. I've updated this comment.

**Pull request merged**